pkg/xds/sync/ingress_proxy_builder.go

```diff
@@ -151,15 +152,16 @@ func (p *IngressProxyBuilder) updateIngress(ctx context.Context, zoneIngress *co
 
 func (p *IngressProxyBuilder) getIngressExternalServices(ctx context.Context) (*core_mesh.ExternalServiceResourceList, error) {
 	meshList := &core_mesh.MeshResourceList{}
-	if err := p.ReadOnlyResManager.List(ctx, meshList, core_store.ListOrdered(), core_store.ListByFilterFunc(func(rs core_model.Resource) bool {
-		return rs.(*core_mesh.MeshResource).ZoneEgressEnabled()
-	})); err != nil {
+	if err := p.ReadOnlyResManager.List(ctx, meshList, store.ListOrdered()); err != nil {
```

Comment:
Should remove the `ListOrdered` from everywhere?

Reply:
I don't think so, because when the cache is disabled we might produce unpredictable config order.

Reply:
Should we disallow disabling the cache?

Reply:
This is a configuration option so we could maybe deprecate it and later remove it. The default is true, but in test, we are creating sometimes ReadOnlyResourceManager without wrapping in the cache and any new test needs to add cache layer/or we could make it somehow default.

Reply:
Yes I think we should make it default but maybe not as part of this PR as it's going to be backported.

Reply:
https://github.com/kumahq/kuma/issues/5584